src/lsdb/dask/merge_catalog_functions.py

```diff
@@ -54,25 +56,51 @@ def concat_partition_and_margin(
     return joined_df
 
 
-def align_catalogs(left: Catalog, right: Catalog) -> PixelAlignment:
+def align_catalogs(left: Catalog, right: Catalog, add_right_margin: bool = True) -> PixelAlignment:
     """Aligns two catalogs, also using the right catalog's margin if it exists
 
     Args:
         left (lsdb.Catalog): The left catalog to align
         right (lsdb.Catalog): The right catalog to align
+        add_right_margin (float): When using MOCs to align catalogs, adds a border to the
+            right catalog's moc to include the margin of the right catalog, if the right margin exists
     Returns:
         The PixelAlignment object from aligning the catalogs
     """
 
+    right_added_radius = None
+
     if right.margin is not None:
         right_tree = align_trees(
             right.hc_structure.pixel_tree,
             right.margin.hc_structure.pixel_tree,
             alignment_type=PixelAlignmentType.OUTER,
         ).pixel_tree
+        if add_right_margin:
+            right_added_radius = right.margin.hc_structure.catalog_info.margin_threshold
     else:
         right_tree = right.hc_structure.pixel_tree
-    return align_trees(left.hc_structure.pixel_tree, right_tree, alignment_type=PixelAlignmentType.INNER)
+
+    right_moc = (
+        right.hc_structure.moc
+        if right.hc_structure.moc is not None
+        else right.hc_structure.pixel_tree.to_moc()
+    )
+    if right_added_radius is not None:
+        right_moc_depth_resol = hp.nside2resol(hp.order2nside(right_moc.max_order), arcmin=True) * 60
+        if right_added_radius < right_moc_depth_resol:
+            right_moc = right_moc.add_neighbours()
+        else:
+            delta_order = int(np.ceil(np.log2(right_added_radius / right_moc_depth_resol)))
+            right_moc = right_moc.degrade_to_order(right_moc.max_order - delta_order).add_neighbours()
```

Comment:
Could we move this auxiliary code to hipscat?

Reply:
Problem is hipscat catalog doesn't have the margin as a parameter. We should refactor a lot more stuff into hipscat I think, so let's open up a ticket to do that.

Reply:
opened #325